### DISCUSSION TITLE
Risk of lock-out when using IP whitelisting for an organization?

### DISCUSSION THREAD

**Discussion by KingOtto123** [community/community#29123] [Enterprise]
We are a small dev team, and are considering GitHub Enterprise Cloud, and using IP whitelisting to ensure that code is only accessed from within our network.

I am trying to understand the impilcations of IP Whitelisting: **Isn't there a very real danger of lock-out for an entire organization when activating IP whitelisting, so that even the org owner becomes unable to access the org and all of its repos?** 

Here is my convern: Say we have 2 cloud clients running on AWS (think of AWS WorkSpaces, or Windows Remote Desktop clients) on which we do development work. Usually, I would expect them to have static IPs. But, if their IP addresses change (for whatever reason), and if GitHub enforces IP compliance for any and all login attemps... Are when then doomed and will never again be able to log in to our organization?

Here is what I understand so far, kindly let me know if I misunderstood something: IP whitelisting also restricts web access, i.e., account will not be able to log in to GitHub website

**Comment by wdpankow** [comment]
There is a genuine possibility you could be locked out of you GitHub organization if your corporate network's IP addresses change. In such an event, the first step would be to **open a ticket** with [GitHub Support](https://support.github.com/). Their primary focus is on maintaining the security of your organization's data, so owner identity verification and permission will be required before they can disable the IP allow list.

There are things you can do to minimize the chances of this happening:
- Have a backup network included in the IP allow list, one that you can rely on to remain static in case your corporate IPs change.
- Maintain at least 2 owners of the organization. This will avoid having a single point of access and may be necessary to approve disabling the allow list.
- Use [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) to add IP address ranges from your network to encompass all available IPs with fewer allow list entries to avoid missing IPs dynamically assigned to hosts in your network.
